### PR TITLE
Basic support for lights and services

### PR DESCRIPTION
**Pull request by marcelveldt**
Basic implementation of a dimmable light (non-color) and some commonly used services.
I will expand these services later on, first need some exploring which ones are really needed.

As for the lights. In the old implementation I see a lot of workarounds and I wonder if these need to be duplicated into the new implementation. I will study the specification some more and create a seperate PR for the color lights. With this PR all lights are treated as basic Zwavedimmers.

Also some very small fixes like PEP formatting and the power_consumption attribute.

I've decided to move the services stuff into a seperate file/class for code hygiene/readability reasons, in the old implementation this was all packed into the config_flow method which is horror to read/follow.

### CODE REVIEW

File: custom_components/zwave_mqtt/services.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+class ZWaveServices:
+    """Class that holds our services ( Zwave Commands) that should be published to hass."""
+
+    def __init__(self, hass, manager):
+        """Initialize with both hass and ozwmanager objects."""
+        self._hass = hass
+        self._manager = manager
+
+    def register(self):
```
Comment (balloob) [inline custom_components/zwave_mqtt/services.py]: Let's make this a @callback and use `self._hass.services.async_register` That way you don't need to call it with `hass.async_add_job`.

File: custom_components/zwave_mqtt/services.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        """Register all our services."""
+        self._hass.services.register(const.DOMAIN, const.SERVICE_ADD_NODE,
+                                     self.add_node, schema=ADD_NODE_SCHEMA)
```
Comment (balloob) [inline custom_components/zwave_mqtt/services.py]: I prefer if we just inline the schema constants. That's the direction Home Assistant is moving towards to.

File: custom_components/zwave_mqtt/__init__.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
 from .const import DATA_NODES, DATA_VALUES, DOMAIN, PLATFORMS, TOPIC_OPENZWAVE
 from .discovery import DISCOVERY_SCHEMAS, check_node_schema, check_value_schema
 from .entity import ZWaveDeviceEntityValues
+from .services import ZWaveServices
```
Comment (balloob) [inline custom_components/zwave_mqtt/__init__.py]: Can we break the services part out into a new PR? That way it's easier to review.

File: custom_components/zwave_mqtt/entity.py [hunk 4]
```diff
@@ -136,9 +137,11 @@ def value_changed(self, value):
         if value.value_id_key in [v.value_id_key for v in self.values if v]:
             self.async_schedule_update_ha_state()
 
+    # pylint: disable=unnecessary-pass
```
Comment (balloob) [inline custom_components/zwave_mqtt/entity.py]: Don't ignore it, just remove the `pass`

File: custom_components/zwave_mqtt/entity.py [hunk 5]
```diff
@@ -149,24 +152,37 @@ def device_info(self):
...
+        if self.power_consumption is not None:
+            attrs[ATTR_POWER] = self.power_consumption
 
         return attrs
 
     @property
     def name(self):
         """Return the name of the entity."""
-        return f"{self.values.primary.node.node_manufacturer_name} {self.values.primary.node.node_product_name}: {self.values.primary.label}"
+        return "%s %s: %s" % (self.values.primary.node.node_manufacturer_name,
```
Comment (balloob) [inline custom_components/zwave_mqtt/entity.py]: Use modern string formatting: `f"{node.node_manufacturer_name} {node.node_product_name} {primary_value.label}"`

File: custom_components/zwave_mqtt/light.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    Light,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .entity import ZWaveDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+# pylint: disable=unnecessary-pass,unused-argument
```
Comment (balloob) [inline custom_components/zwave_mqtt/light.py]: We should globally disable `unused-argument` in `pylintrc`.

File: custom_components/zwave_mqtt/light.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
```
Comment (balloob) [inline custom_components/zwave_mqtt/light.py]: We can remove this, this stuff is all new so there is no need for an "old method".

File: custom_components/zwave_mqtt/light.py [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+            self.values.dimming_duration.send_value(0xFE)
+            _LOGGER.warning("Transition clipped to 127 minutes for %s.", self.entity_id)
+        else:
+            minutes = int(transition / 60)
+            _LOGGER.debug(
+                "Transition rounded to %d minutes for %s.", minutes, self.entity_id
+            )
+            self.values.dimming_duration.send_value(minutes + 0x7F)
+
+    def turn_on(self, **kwargs):
```
Comment (balloob) [inline custom_components/zwave_mqtt/light.py]: This should be `async def async_turn_on`

File: custom_components/zwave_mqtt/light.py [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        # brightness. Level 255 means to set it to previous value.
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = kwargs[ATTR_BRIGHTNESS]
+            brightness = byte_to_zwave_brightness(brightness)
+        else:
+            brightness = 255
+
+        self.values.primary.send_value(brightness)
+
+    def turn_off(self, **kwargs):
```
Comment (balloob) [inline custom_components/zwave_mqtt/light.py]:
```suggestion
    async def async_turn_off(self, **kwargs):
```

File: custom_components/zwave_mqtt/entity.py (hunk 5 above)
Comment (marcelveldt) [reply custom_components/zwave_mqtt/entity.py]: OK, adjusted it to shorten the line. Will put back the modern string formatting with a line break ?

File: custom_components/zwave_mqtt/__init__.py (hunk 3 above)
Comment (marcelveldt) [reply custom_components/zwave_mqtt/__init__.py]: No problem, will do some git sorcery to get those commits into a seperate branch. Thanks for reviewing so far!

**Comment by marcelveldt** [comment]
thanks for reviewing @balloob. I'll close this PR and open 2 separate ones for the light addition and services with you recommended changes.

**Comment by balloob** [comment]
Can you start with services? I am doing some changes to allow adding tests, and I need to make small tweaks to how platforms work.

**Comment by marcelveldt** [comment]
> Can you start with services? I am doing some changes to allow adding tests, and I need to make small tweaks to how platforms work.

just done... see new PR for the services.

**Pull request closed without merging**